File: Startupba/Startupba.Services/Database/Startup.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Startupba.Services.Database
{
    public class Startup
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(4000)]
        public string Description { get; set; } = string.Empty;

        [Required]
        public int FounderId { get; set; } // FK to User (entrepreneur)

        [Required]
        public int CategoryId { get; set; }

        [Required]
        public int CityId { get; set; }

        /// <summary>
        /// Funding goal the founder wants to reach.
        /// </summary>
        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TargetAmount { get; set; }

        /// <summary>
        /// Total of completed donations. Kept up to date when donations complete.
        /// </summary>
        [Column(TypeName = "decimal(18,2)")]
        public decimal AmountRaised { get; set; } = 0;

        /// <summary>
        /// Percentage the platform keeps when the target is reached.
        /// Snapshot of the PlatformSetting value at approval time (transparency for the founder).
        /// </summary>
        [Column(TypeName = "decimal(5,2)")]
        public decimal PlatformFeePercent { get; set; } = 0;

        /// <summary>
        /// 1=Draft, 2=Pending, 3=Approved, 4=Rejected, 5=Paused, 6=Completed
        /// </summary>
        [Required]
        public int StatusId { get; set; } = StartupStatuses.Pending;

        /// <summary>
        /// Reason the admin gave when rejecting the startup.
        /// </summary>
        [MaxLength(1000)]
        public string? RejectionReason { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public DateTime? UpdatedAt { get; set; }

        public DateTime? ApprovedAt { get; set; }

        public DateTime? CompletedAt { get; set; }

        // Navigation properties
        public User Founder { get; set; } = null!;
        public Category Category { get; set; } = null!;
        public City City { get; set; } = null!;
        public StartupStatus Status { get; set; } = null!;

        public ICollection<StartupImage> StartupImages { get; set; } = new List<StartupImage>();
        public ICollection<Donation> Donations { get; set; } = new List<Donation>();
        public ICollection<StartupLike> StartupLikes { get; set; } = new List<StartupLike>();
        public ICollection<Favorite> Favorites { get; set; } = new List<Favorite>();
        public ICollection<BlogPost> BlogPosts { get; set; } = new List<BlogPost>();
    }
}

File: Startupba/UI/startupba_desktop/lib/model/startup.dart
import 'package:json_annotation/json_annotation.dart';

part 'startup.g.dart';

@JsonSerializable()
class Startup {
  final int id;
  final String name;
  final String description;
  final int founderId;
  final String founderName;
  final int categoryId;
  final String categoryName;
  final int cityId;
  final String cityName;
  final double targetAmount;
  final double amountRaised;
  final double fundingPercent;
  final double platformFeePercent;
  final int statusId;
  final String statusName;
  final String? rejectionReason;
  final bool isActive;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final DateTime? approvedAt;
  final DateTime? completedAt;
  final int likeCount;
  final int favoriteCount;
  final int donationCount;
  final String? coverImage;
  final String? logoImage;

  Startup({
    this.id = 0,
    this.name = '',
    this.description = '',
    this.founderId = 0,
    this.founderName = '',
    this.categoryId = 0,
    this.categoryName = '',
    this.cityId = 0,
    this.cityName = '',
    this.targetAmount = 0,
    this.amountRaised = 0,
    this.fundingPercent = 0,
    this.platformFeePercent = 0,
    this.statusId = 0,
    this.statusName = '',
    this.rejectionReason,
    this.isActive = true,
    required this.createdAt,
    this.updatedAt,
    this.approvedAt,
    this.completedAt,
    this.likeCount = 0,
    this.favoriteCount = 0,
    this.donationCount = 0,
    this.coverImage,
    this.logoImage,
  });

  factory Startup.fromJson(Map<String, dynamic> json) =>
      _$StartupFromJson(json);
  Map<String, dynamic> toJson() => _$StartupToJson(this);
}
